### PR TITLE
[FIX] *: python 3.8 compatibility

### PR DESCRIPTION
**Pull request by mart-e**
Remove warnings
Add missing parameter posonlyargs (cf [position swap](https://docs.python.org/3/whatsnew/changelog.html#python-3-8-0-beta-3))

### CODE REVIEW

File: odoo/addons/base/models/qweb.py [hunk 1]
```diff
@@ -311,7 +312,8 @@ def compile(self, template, options):
         try:
             # noinspection PyBroadException
             ns = {}
-            unsafe_eval(compile(astmod, '<template>', 'exec'), ns)
+            compres = compile(astmod, '<template>', 'exec')
+            unsafe_eval(compres, ns)
```
Comment (xmo-odoo) [inline odoo/addons/base/models/qweb.py]: Maybe no?

Comment (mart-e) [reply odoo/addons/base/models/qweb.py]: oh right

**Comment by xmo-odoo** [comment]
Also not sure the `is not` -> `!=` quite belong in here.

File: odoo/addons/base/models/qweb.py [hunk 2]
```diff
@@ -311,7 +312,7 @@ def compile(self, template, options):
         try:
             # noinspection PyBroadException
             ns = {}
-            unsafe_eval(compile(astmod, '<template>', 'exec'), ns)
+            unsafe_eval(comprescompile(astmod, '<template>', 'exec'), ns)
```
Comment (xmo-odoo) [inline odoo/addons/base/models/qweb.py]: I don't think comprescompile is a proper python builtin
```suggestion
            unsafe_eval(compile(astmod, '<template>', 'exec'), ns)
```

**Comment by mart-e** [comment]
> Also not sure the is not -> != quite belong in here.

No it does, otherwise you get a warning in the logs
```
WARNING ? py.warnings: .../odoo/addons/base/models/res_config.py:600: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if method.startswith('set_') and method is not 'set_values':
```

**Comment by xmo-odoo** [comment]
> No it does, otherwise you get a warning in the logs

TIL 3.8 adds lints in core.

**Comment by xmo-odoo** [comment]
robodoo r+

**Comment by mart-e** [comment]
@odony @antonylesuisse should we backport this to 11.0?

**Comment by odony** [comment]
> @odony @antonylesuisse should we backport this to 11.0?

What are the consequences? Extra warnings in the logs? Seems really minor, I wouldn't bother backporting, since we're not building any branch against 3.8. Not even sure why this is in 13.0 and not just master, to be honest.

BTW it would be nice to explain the issue in details, since there is no linked build log. What kind of warnings are we talking about? The `is not` one was mentioned in the comments, but not in the commit message.
And I don't know why we suddenly need to pass that `posonlyargs`, especially when "*ast.arguments does no validation whatsoever*". What's the issue?

Bonus: are we going to have 3.8 builds to ensure we won't be breaking this again, or extra pylint rules in master, or we just hope we won't break this again? :p

Thanks ;-)

**Comment by xmo-odoo** [comment]
> And I don't know why we suddenly need to pass that posonlyargs, especially when "ast.arguments does no validation whatsoever". What's the issue?

It doesn't matter that it's added in 3.6 or 3.7 (it's ignored), but it doesn't work if it's not present in 3.8.

**Comment by mart-e** [comment]
You are right, the commit message was not fully describing the issue. In 3.8, when trying to access the login page, got an error:
```
Traceback (most recent call last):
  File "/home/mat/odoo/odoo/odoo/addons/base/ir/ir_qweb/qweb.py", line 329, in compile
    unsafe_eval(compile(astmod, '<template>', 'exec'), ns)
TypeError: required field "posonlyargs" missing from arguments

Error when compiling AST
TypeError: required field "posonlyargs" missing from arguments
```
Removing the warning can be skipped in 11.0.

**Pull request closed without merging**